### PR TITLE
OCL OSD implementation and VPP OCL framework changes

### PR DESCRIPTION
**Pull request by dspmeng**
ocl: osd with contrastive font color
 * osd font color is black or white contrastive to background

ocl: support cl program with multiple kernels

ocl: implement ocl vpp image wrappers
  * OclVppRawImage wraps cpu accesible memory to hand unaligned area
  * OclVppCLImage wraps VASurface backed ocl image object

### CODE REVIEW

**Comment by dspmeng** [comment]
Modify patches according review comments

**Comment by dspmeng** [comment]
update font.h

File: interface/VideoPostProcessParams.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301 USA
+ */
+
+#ifndef __VIDEO_POST_PROCESS_PARAMS_H__
```
Comment (xuguangxin) [inline interface/VideoPostProcessParams.h]: should follow other name just as VideoEncoderDefs.h

File: vpp/vaapipostprocess_base.h [hunk 2]
```diff
@@ -42,6 +42,7 @@ class VaapiPostProcessBase : public IVideoPostProcess {
     // for some type of vpp such as deinterlace, we will hold a referece of src.
     virtual YamiStatus process(const SharedPtr<VideoFrame>& src,
                                const SharedPtr<VideoFrame>& dest) = 0;
+    virtual YamiStatus setParameters(VppParamType type, void* vppParam) {return YAMI_SUCCESS;}
```
Comment (xuguangxin) [inline vpp/vaapipostprocess_base.h]: showed return invalid param for this

**Comment by xuguangxin** [comment]
almost ok for me.
But this is large commit please use https://llvm.org/svn/llvm-project/cfe/trunk/tools/clang-format/git-clang-format
with --style=webkit to format your patchs

**Comment by dspmeng** [comment]
1. rename interface/VideoPostProcessParams.h to interface/VideoPostProcessDefs.h
2. return YAMI_INVALID_PARAM in VaapiPostProcessBase::setParameters()
3. reformat patches with git-clang-format --style=webkit

**Comment by dspmeng** [comment]
update osd patch: remove font block width parameter, and use crop height instead.

File: interface/VideoPostProcessDefs.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (xuguangxin) [inline interface/VideoPostProcessDefs.h]: seems it not install in $install_dir/include/libyami. seems you need add it libyami_vpp_source_h

File: interface/VideoPostProcessInterface.h [hunk 4]
```diff
@@ -41,6 +42,7 @@ class IVideoPostProcess {
     // for some type of vpp such as deinterlace, we will hold a referece of src.
     virtual YamiStatus process(const SharedPtr<VideoFrame>& src,
                                const SharedPtr<VideoFrame>& dest) = 0;
+    virtual YamiStatus setParameters(VppParamType type, void* vppParam) = 0;
```
Comment (xuguangxin) [inline interface/VideoPostProcessInterface.h]: it change interface. please update yami_api_micro_version in configure.ac

**Comment by dspmeng** [comment]
update osd patch per latest review comments. please review.

**Comment by xuguangxin** [comment]
+1

**Comment by xuguangxin** [comment]
merged. thanks mengjia

**Pull request merged**